packages/pigeon/lib/java_generator.dart

```diff
@@ -301,24 +279,53 @@ static MessageCodec<Object> getCodec() {
   });
 }
 
-String _makeGetter(Field field) {
+String _makeGetter(NamedType field) {
   final String uppercased =
       field.name.substring(0, 1).toUpperCase() + field.name.substring(1);
   return 'get$uppercased';
 }
 
-String _makeSetter(Field field) {
+String _makeSetter(NamedType field) {
   final String uppercased =
       field.name.substring(0, 1).toUpperCase() + field.name.substring(1);
   return 'set$uppercased';
 }
 
-String? _javaTypeForDartType(String datatype) {
-  return _javaTypeForDartTypeMap[datatype];
+/// Converts a [List] of [TypeDeclaration]s to a comma separated [String] to be
+/// used in Java code.
+String _flattenTypeArguments(List<TypeDeclaration> args) {
+  return args
+      .map((TypeDeclaration e) => _javaTypeForDartType(e) ?? e.dataType)
+      .reduce((String value, String element) => '$value, $element');
+}
+
+String? _javaTypeForDartType(TypedEntity field) {
```

Comment:
s/field/entity/ (or "type" since you're already calling it a type in the function name).

Reply:
done